pkg/rabbit/setup.go

```diff
@@ -0,0 +1,132 @@
+/*
+Copyright 2022 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package rabbit
+
+import (
+	"errors"
+	"time"
+
+	"github.com/NeowayLabs/wabbit"
+	wabbitamqp "github.com/NeowayLabs/wabbit/amqp"
+	"go.uber.org/zap"
+)
+
+const cycleRetries = 60
+
+var (
+	retryCounter                                        = 0
+	cycleNumber                                         = 0
+	InitCycleDuration                                   = 1
+	cycleDuration                                       = 1
+	maxCycleDuration                                    = 7200
+	Retrying                                            = true
+	cleaningUp                                          = false
+	DialFunc          func(string) (wabbit.Conn, error) = wabbitamqp.Dial
+)
+
+func SetDialFunc(dialFunc func(uri string) (wabbit.Conn, error)) {
+	DialFunc = dialFunc
+}
+
+func SetupRabbitMQ(
+	brokerURL string,
```

Comment:
nit: brokerURL is confusing here since we have a different brokerURL that's basically the ingress service URL. Maybe RabbitMQURL or ServerURL